lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php

```diff
@@ -378,18 +389,14 @@ public function clearAllCache(\DateTime $date = null)
     {
     }
 
-    private function createTermQueryParams($query, SearchEngineOptions $options)
+    private function highlightFields(array $fields, $terms)
     {
-        $params = [
-            'index' => $this->indexName,
-            'type'  => TermIndexer::TYPE_NAME,
-            'body'  => [],
-            'size'  => 20,
-        ];
-
-        $params['body']['query'] = $query;
+        $highlight = [];
+        foreach ($fields as $field) {
+            $highlight[$field] = ['highlight_query' => ['bool' => ['should' => ['match' => [$field => ['query' => $terms]]]]]];
```

Comment:
`$highlight[$field]['highlight_query']['match'][$field]['query'] = $terms;` is simpler.